asdf/_asdf.py

```diff
@@ -732,6 +732,41 @@ def get_array_compression_kwargs(self, arr):
         """ """
         return self._blocks._get_array_compression_kwargs(arr)
 
+    def set_array_save_base(self, arr, save_base):
```

Comment:
Yes, this option does not get explicitly written to the file (similar to the `array_compression_kwargs`). However the setting does "roundtrip" because disables `array_save_base` on a view will cause it to get a new block (and then read in as a non-view array).

For a tree with let's say 3 arrays:
- `big_array`
- `view_of_big_array`
- `other_array`

Writing this tree (with the defaults) will create:
- 1 block for `big` and `view`
- 1 block for `other`

When read back in `view` will become a view and rewriting the file will produce the same result.

If however the user chooses to call `set_array_save_base(view, False)` the file will write out with 3 blocks
- 1 block for `big`
- 1 block for `view`
- 1 block for `other
When this file is read back in the user will get a tree where `view` is no longer a view (due to the `set_array_save_base` call) and writing it back out will again produce a file with 3 blocks.

This only works because our current default is `True`. If we change the default to `False` than "viewness" will not be preserved and this setting would have to be re-set every time a file is written.

The thought behind making this per-object was to allow a user to have more flexibility. For example, let's say the tree contains:
- `wcs`: wcs with possibly many arrays, some might be views of non-tree arrays and we don't want these to be views
- `sci`: science data, a big non-view array
- `thumbnail`: cutout of science data that we want to keep as a view of "science"
- `ref`: reference pixels from the science data, where for record-keeping reasons I want this to be a non-view on write

If we were to provide only an `AsdfConfig` option the user would have to copy `ref` and `wcs` to keep `thumbnail` as a view.